src/core/cpu_patches.cpp

```diff
@@ -307,9 +307,24 @@ static void GenerateBLSI(const ZydisDecodedOperand* operands, Xbyak::CodeGenerat
 
     SaveRegisters(c, {scratch});
 
+    // BLSI sets CF to zero if source is zero, otherwise it sets CF to one.
+    Xbyak::Label carrySet, carryNotSet, end;
+
     c.mov(scratch, *src);
-    c.neg(scratch);
+    c.neg(scratch); // NEG, like BLSI, clears CF if the source is zero and sets it otherwise
+    c.jc(carrySet);
+    c.jmp(carryNotSet);
+
+    c.L(carrySet);
     c.and_(scratch, *src);
+    c.stc(); // setting/clearing carry needs to happen after the AND because that clears CF
+    c.jmp(end);
+
+    c.L(carryNotSet);
+    c.and_(scratch, *src);
+    c.clc();
```

Comment:
This CLC is redundant given that and has already cleared CF, but if you want to abuse this you should definitely put a comment about it here.

There's some easy ways to copy ZF to CF without any branches but the AND makes it fairly annoying

Reply:
Oh true thanks

Yeah I tried to think of a way to do it without branches and couldn't come up with something